### PR TITLE
Apply some of Gerry's best practices.  Balance readability and minima…

### PR DESCRIPTION
**Pull request by dukejones**
…l layers.

Read through the docker best practices guide.  Really good, also I'm on board with running services as a non-privileged user in containers now.  (Which, assuming using this as a base image, would be good to set up in this image for DNA containers that use this as a base.)

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.git
+examples
```
Comment (GerryG) [inline .dockerignore]: I should use .dockerignore more often, thx for the reminder.

File: Dockerfile [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
+    && cd /app/golang/src/github.com/metacurrency/holochain \
+    && make deps \
+    && chown -R holochain /app
 
-RUN  go get -v -d github.com/metacurrency/holochain
+WORKDIR /app/golang/src/github.com/metacurrency/holochain
 
-WORKDIR $GOPATH/src/github.com/metacurrency/holochain
-RUN make deps
+USER holochain
```
Comment (GerryG) [inline Dockerfile]: Ah, good idea. Should use this feature more too. 

Can you still get a root shell in the container when you do this? I'm thinking you would need a root pw or other means (sudo) of getting root if this is set in the image.

File: Dockerfile [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
 
-ADD .  $GOPATH/src/github.com/metacurrency/holochain
-WORKDIR $GOPATH/src/github.com/metacurrency/holochain
+COPY . /app/golang/src/github.com/metacurrency/holochain
```
Comment (GerryG) [inline Dockerfile]: Ahh, use copy. One point that may be just to be aware of. You might have files left over from the released version that was here already. Any files that got removed, that is. Easily fixed by commiting the deletes.

File: Dockerfile (hunk 2 above)
Comment (dukejones) [inline Dockerfile]: Good question! Let me know if you find out, I'll look also.

File: Dockerfile (hunk 3 above)
Comment (dukejones) [inline Dockerfile]: good point -- i'll add a `rm -rf`

File: Dockerfile (hunk 2 above)
Comment (GerryG) [inline Dockerfile]: I tested this and it seemed like not. I added a PR that gives this option back.

**Pull request merged**